pypi/pip_generate.go

```diff
@@ -172,17 +172,17 @@ func pyPIToBazelPackageName(packageName string) string {
 }
 
 // Takes a PyPI dependency and returns just the package name part, without extras
-func dependencyPackageName(dependency string) string {
+func dependencyNormalizedPackageName(dependency string) string {
 	// PyPI packages can contain upper case characters, but are matched insensitively
-	dependency = strings.ToLower(dependency)
+	packageName := strings.ToLower(dependency)
 	// PyPI packages contain -, but the wheel and bazel names convert them to _
-	dependency = strings.Replace(dependency, "-", "_", -1)
+	packageName = strings.Replace(packageName, "-", "_", -1)
 
-	extraStart := strings.IndexByte(dependency, '[')
+	extraStart := strings.IndexByte(packageName, '[')
 	if extraStart >= 0 {
-		return dependency[:extraStart]
+		packageName = packageName[:extraStart]
 	}
-	return dependency
+	return normalizePackageName(packageName)
```

Comment:
I think we can delete line 177 since this is also going to call `strings.ToLower()`?